cg/meta/workflow/balsamic.py

```diff
@@ -236,6 +243,24 @@ def get_verified_tumor_path(sample_data: dict) -> str:
             )
         return tumor_paths[0]
 
+    @staticmethod
+    def get_verified_gender(sample_data: dict) -> str:
+        """Takes a dict with samples and attributes, and retrieves the case gender"""
+
+        genders = []
+        samples = []
+        for key, val in sample_data.items():
+            samples.append(key)
+            genders.append(val["gender"])
+
+        unique_gender = all(gender == genders[0] for gender in genders)
+
+        if genders and unique_gender and genders[0] in [Gender.FEMALE, Gender.MALE]:
+            return genders[0]
+        else:
+            LOG.error(f"Unable to retrieve a valid gender from samples: {samples}")
+            raise BalsamicStartError
+
```

Comment:
Maybe this could be made more readable, something along the lines of:

```suggestion
    @staticmethod
    def get_verified_gender(samples_data: dict) -> Union[Gender.FEMALE, Gender.MALE]:
        """Takes a dict with samples and attributes, and retrieves the case gender"""

        gender: str
        genders: Set = set()
        for sample_data in samples_data.values():
            gender = sample_data["gender"]
            genders.append(gender)

        is_one_gender = len(genders) == 1

        if is_one_gender and gender in [Gender.FEMALE, Gender.MALE]:
            return gender
        
        LOG.error(f"Unable to retrieve a valid gender from samples: {samples_data.keys()}")
        raise BalsamicStartError
```

Reply:
Thank you for your review @patrikgrenfeldt ⭐ !

I think I managed to simplify the code a little bit. You can check the changes here and tell me what you think: [3e60993](https://github.com/Clinical-Genomics/cg/pull/1493/commits/3e6099310b9aad91bfc2303781295e015b8b389b)